src/NuGet.Core/NuGet.Packaging/PackageExtraction/PackageHelper.cs

```diff
@@ -186,49 +175,66 @@ public static IEnumerable<ZipFilePair> GetInstalledPackageFiles(IEnumerable<ZipF
         /// <param name="packageSaveMode"></param>
         /// <param name="token"></param>
         /// <returns></returns>
-        public static async Task<IEnumerable<ZipFilePair>> GetInstalledPackageFiles(Stream packageStream,
+        public static IEnumerable<ZipFilePair> GetInstalledPackageFiles(Stream packageStream,
             PackageIdentity packageIdentity,
             PackagePathResolver packagePathResolver,
-            PackageSaveModes packageSaveMode,
-            CancellationToken token)
+            PackageSaveModes packageSaveMode)
         {
-            var installedPackageFiles = new List<ZipFilePair>();
+            var installedPackageFiles = Enumerable.Empty<ZipFilePair>();
             var packageDirectory = packagePathResolver.GetInstalledPath(packageIdentity);
             if (!String.IsNullOrEmpty(packageDirectory))
             {
                 var zipArchive = new ZipArchive(packageStream);
-                var packageFiles = await GetPackageFiles(zipArchive.Entries, packageDirectory, packageSaveMode, token);
-                installedPackageFiles.AddRange(GetInstalledPackageFiles(packageFiles));
+                var packageFiles = EnumeratePackageFiles(zipArchive.Entries, packageDirectory, packageSaveMode);
+                installedPackageFiles = GetInstalledPackageFiles(packageFiles);
             }
 
-            return installedPackageFiles;
+            return installedPackageFiles.ToList();
         }
 
-        public static async Task<Tuple<string, IEnumerable<ZipFilePair>>> GetInstalledSatelliteFiles(Stream packageStream,
+        public static Tuple<string, IEnumerable<ZipFilePair>> GetInstalledSatelliteFiles(Stream packageStream,
```

Comment:
it is whole lot of refactoring as it's deeply integrated with several methods in a call chain.
I'd prefer to avoid doing it unless it is really needed.